bin/main.rs

```diff
@@ -1,3 +1,372 @@
+        env = "WADM_EVENT_STREAM"
+    )]
+    event_stream_name: String,
+
+    /// Name of the commands stream to use
+    #[arg(
+        long = "command-stream",
+        default_value = "wadm_commands",
+        env = "WADM_COMMAND_STREAM"
+    )]
+    command_stream_name: String,
+
+    /// The NATS JetStream domain to connect to
+    #[arg(short = 'd', env = "WADM_JETSTREAM_DOMAIN")]
+    domain: Option<String>,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    configure_tracing(
+        args.structured_logging,
+        args.tracing_enabled,
+        args.tracing_endpoint,
+    );
+
+    // TODO: All the NATS connection options and jetstream stuff
+    let client = async_nats::connect("127.0.0.1:4222").await?;
+    let context = if let Some(domain) = args.domain {
+        jetstream::with_domain(client.clone(), domain)
+    } else {
+        jetstream::new(client.clone())
+    };
+
+    let event_stream = context
+        .get_or_create_stream(Config {
+            name: args.event_stream_name,
+            description: Some(
+                "A stream that stores all events coming in on the wasmbus.evt topics in a cluster"
+                    .to_string(),
+            ),
+            num_replicas: 1,
+            retention: async_nats::jetstream::stream::RetentionPolicy::WorkQueue,
+            subjects: vec![DEFAULT_EVENTS_TOPIC.to_owned()],
+            max_age: DEFAULT_EXPIRY_TIME,
+            storage: async_nats::jetstream::stream::StorageType::File,
+            allow_rollup: false,
+            ..Default::default()
+        })
+        .await
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    let command_stream = context
+        .get_or_create_stream(Config {
+            name: args.command_stream_name,
+            description: Some("A stream that stores all commands for wadm".to_string()),
+            num_replicas: 1,
+            retention: async_nats::jetstream::stream::RetentionPolicy::WorkQueue,
+            subjects: vec![DEFAULT_COMMANDS_TOPIC.to_owned()],
+            max_age: DEFAULT_EXPIRY_TIME,
+            storage: async_nats::jetstream::stream::StorageType::File,
+            allow_rollup: false,
+            ..Default::default()
+        })
+        .await
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    let host_id = args
+        .host_id
+        .unwrap_or_else(|| uuid::Uuid::new_v4().to_string());
+    tokio::select! {
+        res = handle_events(event_stream, context.clone(), host_id) => {
+            res?
+        }
+        res = handle_commands(command_stream) => {
+            res?
+        }
+    }
+    Ok(())
+}
+
+// NOTE: These functions will probably need to be replaced with something in the library code, but
+// they're good enough for testing
+async fn handle_events(stream: Stream, context: Context, host_id: String) -> anyhow::Result<()> {
+    let mut consumer = EventConsumer::new(stream, "wasmbus.evt.default")
```

Comment:
assuming this hard coded topic is just because this is a test/experiment?

Reply:
Yes, 100% hard coded. If you take a look at #61 you'll see that I made something to support multiple.